subscriptions/subscriptions-impl/src/test/java/com/duckduckgo/subscriptions/impl/rmf/RMFPProDaysSinceSubscribedMatchingAttributeTest.kt

```diff
@@ -0,0 +1,227 @@
+package com.duckduckgo.subscriptions.impl.rmf
+
+import com.duckduckgo.common.test.CoroutineTestRule
+import com.duckduckgo.common.utils.CurrentTimeProvider
+import com.duckduckgo.remote.messaging.api.JsonMatchingAttribute
+import com.duckduckgo.subscriptions.api.SubscriptionStatus.AUTO_RENEWABLE
+import com.duckduckgo.subscriptions.impl.SubscriptionsManager
+import com.duckduckgo.subscriptions.impl.repository.Entitlement
+import com.duckduckgo.subscriptions.impl.repository.Subscription
+import java.util.concurrent.TimeUnit
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+import org.mockito.kotlin.whenever
+
+class RMFPProDaysSinceSubscribedMatchingAttributeTest {
+    @get:Rule
+    val coroutineTestRule: CoroutineTestRule = CoroutineTestRule()
+
+    @Mock
+    private lateinit var subscriptionsManager: SubscriptionsManager
+
+    @Mock
+    private lateinit var currentTimeProvider: CurrentTimeProvider
+
+    private lateinit var matcher: RMFPProDaysSinceSubscribedMatchingAttribute
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.openMocks(this)
+        matcher = RMFPProDaysSinceSubscribedMatchingAttribute(
+            subscriptionsManager,
+            currentTimeProvider,
+        )
+    }
+
+    @Test
+    fun whenMapKeyIsPProDaysSinceSubscribedThenReturnMatchingAttribute() = runTest {
+        val jsonMatchingAttribute = JsonMatchingAttribute(min = 20, max = 30)
+        val result = matcher.map("pproDaysSinceSubscribed", jsonMatchingAttribute)
+        assertEquals(PProDaysSinceSubscribedMatchingAttribute(min = 20, max = 30), result)
+    }
+
+    @Test
+    fun whenMapKeyIsNotPProDaysSinceSubscribedTheMapReturnNull() = runTest {
+        val jsonMatchingAttribute = JsonMatchingAttribute(min = 20, max = 30)
+        val result = matcher.map("somethingelse", jsonMatchingAttribute)
+        assertNull(result)
+    }
+
```

Comment:
To add a test for the invalid case where we receive value with a non Int type. It should return null or exception.